scripts/golint.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+set -ex
+
+make tidy
+make build
+
+# intentionally capture stderr, so status-errors are also PR-failing.
+# in particular this catches "dubious ownership" failures, which otherwise
+# do not fail this check and the $() hides the exit code.
+if [ -n "$(git status --porcelain  2>&1)" ]; then
```

Comment:
nit: generally prefer `[[` and it's not needed, but also set -o pipefail

Reply:
I'm _pretty_ sure I do still need to capture output with `$()`, and bash / `[[` don't change this semantic in that sense... but this is more than niche enough that I haven't actually tried it so now I'm curious.  let's see what happens.

Reply:
... and I'm struggling to reproduce exactly what I thought I remembered 🤔 

gonna just stick with what the server does here since it's reasonably well proven over there, and also seems to work correctly here.

bash + pipefail does seem promising enough to check at another time though.